XenCenterLib/NamedPipes.cs

```diff
@@ -266,7 +266,39 @@ public Pipe(string path)
             public static bool ExistsPipe(string pipePath)
             {
                 log.Debug($"Checking {pipePath} exists");
-                return Directory.GetFiles(@"\\.\pipe\").Contains(pipePath);
+
+                // CA-382850: We iterate manually in order to catch ArgumentExceptions
+                // when listing files. Pipes can be created with invalid characters in 
+                // their names. This throws exception when those files are accessed.
+                // Other processes might create these pipes and inadvertently prevent
+                // XenCenter from starting
+
+                var e = Directory.EnumerateFiles(@"\\.\pipe\");
+                using (var enumerator = e.GetEnumerator())
+                {
+                    while (true)
+                    {
+                        try
+                        {
+                            if (!enumerator.MoveNext())

```

Comment:
Does movenext() cause the invalid characters exception to trigger? 
If not then you can change the loop so its while(enumerator.movenext())

Reply:
yeah it throws it when the enumerator yields back the file, that's why I couldn't do a simple `foreach` + character check here :(